tests/mock-snapd.c

```diff
@@ -1507,6 +1521,17 @@ mock_snap_get_disabled (MockSnap *snap)
     return snap->disabled;
 }
 
+void
+mock_snap_set_donation (MockSnap *snap, const GPtrArray *donation)
+{
+    g_ptr_array_set_free_func(snap->donation, g_free);
```

Comment:
The copy is not required - in general we'd just reference `donation` here.

Reply:
Can you kindly explain this a bit more? You mean I don't need to copy the links from the `donation` variable into the `snap->donation`?

Reply:
This can be `g_ptr_array_unref(snap->donation); snap->donation = g_ptr_array_ref(donation)`, see nearby functions for the same pattern.